### PR TITLE
Consistently parse all boolean environment variables

### PR DESCRIPTION
**Pull request by giordano**
Note: this is basically only checking the truthy values, so values like `"truw"` would silently be parsed as a falsy value, but as far as I can tell that's what's happening at the moment anyways, so I don't think there is any practical change.

Address https://github.com/JuliaLang/Pkg.jl/issues/2705#issuecomment-978147571.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
I think there's some spots in PlatformEngines where we do this too. Might want to allow passing whether the default should be true or false? Although if the default is always false, I guess no need.

**Comment by giordano** [comment]
> I think there's some spots in PlatformEngines where we do this too.

Oh yes, I saw
https://github.com/JuliaLang/Pkg.jl/blob/0d6a726e91a6a29b3f47fca2396ca5ea7fa5511b/src/PlatformEngines.jl#L228-L231
but that uses a three-value logic,  I wasn't sure whether to add that to the helper function `bool_env`, it'd require all other places to guard `nothing` values.

**Comment by giordano** [comment]
Bump :slightly_smiling_face:

File: src/Artifacts.jl [hunk 1]
```diff
@@ -357,7 +358,7 @@ function download_artifact(
             # Since tree hash calculation is still broken on some systems, e.g. Pkg.jl#1860,
             # and Pkg.jl#2317 so we allow setting JULIA_PKG_IGNORE_HASHES=1 to ignore the
             # error and move the artifact to the expected location and return true
-            ignore_hash = get(ENV, "JULIA_PKG_IGNORE_HASHES", nothing) == "1"
+            ignore_hash = bool_env("JULIA_PKG_IGNORE_HASHES")
```
Comment (IanButterworth) [inline src/Artifacts.jl:361]: Perhaps
```suggestion
            ignore_hash = get_bool_env("JULIA_PKG_IGNORE_HASHES")
```

Comment (giordano) [reply src/Artifacts.jl:361]: Not sure what you mean, is this a suggestion to rename the function?

Comment (IanButterworth) [reply src/Artifacts.jl:361]: Yeah. Just a minor suggestion

**Review by IanButterworth — APPROVED** [review]

**Comment by IanButterworth** [comment]
I think we'd want env vars that exist for shared functionality in both LTS and new releases to behave the same, so I'm going to mark this for backport

**Pull request merged**